### ISSUE TITLE
False positive when overriding __module__

### ISSUE THREAD

**Issue by tbekolay** [PyCQA/pylint#2451]
### Steps to reproduce

Run pylint on the following file:

```python
class ObjectProxy:
    @property
    def __module__(self):
        return "actual.module"

    @property
    def __doc__(self):
        return "Docstring"
```

### Current behavior

Among the correct issues, this one is listed:

```
pylint.py:3:4: E0102: method already defined line 1 (function-redefined)
```

Another minor (but perhaps related) issue is that this issue is also listed:

```
pylint.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)
```

### Expected behavior

I would expect overriding `__module__` to work the same way as overriding `__doc__`, in that doing so doesn't fail the `function-redefined` and contributes to the list of public methods.

This came up because we implement a minimal version of [wrapt](https://wrapt.readthedocs.io/en/latest/). The class that implements `__module__` is a decorator, and `__module__` is overwritten so that it reflects the decorated class, rather than the class decorator.

This didn't used to come up, so it likely is related to the somewhat recent #474.

### pylint --version output

```
pylint 2.1.1
astroid 2.0.4
Python 3.6.3 (default, Oct  3 2017, 21:16:13) 
[GCC 7.2.0]
```

**Comment by PCManticore** [comment]
Hey thanks for creating an issue! Definitely, this should not happen.

**Comment by tbekolay** [comment]
Thanks for the quick fix!
